src/EulerHS/KVConnector/Flow.hs

```diff
@@ -454,48 +434,50 @@ updateObjectRedis meshCfg updVals addPrimaryKeyToWhereClause whereClause obj = d
+          newSkeys =  map (\(SKey s) -> s) (secondaryKeys table)
+          (modifiedSkeysNew, _) = applyFPair (map (getSortedKeyAndValue tName)) $
+                                                segregateList (`isKeyModified` updValsMap) newSkeys
       mapRight (const table) <$> runExceptT (do
-                                    mapM_ ((ExceptT . resetTTL) . (fromString . T.unpack)) unModifiedSkeys
-                                    mapM_ (ExceptT . addNewSkey pKey tName) (foldSkeysFunc modifiedSkeysValuesMap newSkeysValues))
+                                    mapM_ (ExceptT . resetTTL) unModifiedSkeys
+                                    mapM_ (ExceptT . deletePkeyFromSkey pKey) modifiedSkeysOld
+                                    mapM_ (ExceptT . addPkeyToSkey pKey) modifiedSkeysNew)
 
-    resetTTL key= do
-      x <- L.runKVDB meshCfg.kvRedis $ L.expire key meshCfg.redisTtl
+    resetTTL Nothing = pure $ Right False
+    resetTTL (Just key) = do
+      x <- L.rExpire meshCfg.kvRedis (fromString $ T.unpack key) meshCfg.redisTtl
       pure $ mapLeft MRedisError x
+    
+    deletePkeyFromSkey _ Nothing = pure $ Right 0
+    deletePkeyFromSkey pKey (Just key) = mapLeft MRedisError <$> L.sRemB meshCfg.kvRedis (fromString $ T.unpack key) [pKey]
 
-    foldSkeysFunc :: HashMap Text Text -> [(Text, Text)] -> [(Text, Text, Text)]
-    foldSkeysFunc _ [] = []
-    foldSkeysFunc hm (x : xs) = do
-      case HM.lookup (fst x) hm of
-        Just val -> (fst x, snd x, val) : foldSkeysFunc hm xs
-        Nothing -> foldSkeysFunc hm xs
+    addPkeyToSkey :: ByteString -> Maybe Text -> m (MeshResult Bool)
+    addPkeyToSkey _ Nothing = pure $ Right False
+    addPkeyToSkey pKey (Just key) = do
+      _ <- L.rSadd meshCfg.kvRedis (fromString $ T.unpack key) [pKey]
+      mapLeft MRedisError <$> L.rExpireB meshCfg.kvRedis (fromString $ T.unpack key) meshCfg.redisTtl
 
-
-    addNewSkey :: ByteString -> Text -> (Text, Text, Text) -> m (MeshResult ())
-    addNewSkey pKey tName (k, v1, v2) = do
-      let newSKey = fromString . T.unpack $ tName <> "_" <> k <> "_" <> v1
-          oldSKey = fromString . T.unpack $ tName <> "_" <> k <> "_" <> v2
-      res <- runExceptT $ do
-        _ <- ExceptT $ L.runKVDB meshCfg.kvRedis $ L.srem oldSKey [pKey]
-        _ <- ExceptT $ L.runKVDB meshCfg.kvRedis $ L.sadd newSKey [pKey]
-        ExceptT $ L.runKVDB meshCfg.kvRedis $ L.expire newSKey meshCfg.redisTtl
-      case res of
-        Right _ -> pure $ Right ()
-        Left err -> pure $ Left (MRedisError err)
-
-    getSortedKeyAndValue :: [(Text,Text)] -> (Text, Text)
-    getSortedKeyAndValue kvTup = do
+    getSortedKeyAndValue :: Text -> [(Text,Text)] -> Maybe Text
+    getSortedKeyAndValue tName kvTup = do
       let sortArr = sortBy (compare `on` fst) kvTup
       let (appendedKeys, appendedValues) = applyFPair (T.intercalate "_") $ unzip sortArr
-      (appendedKeys, appendedValues)
+      if length (filter (\(_, v) -> v=="") kvTup) > 0
```

Comment:
`not $ null` is better rather than `length > 0`

Reply:
changed